src/components/MyKiva/BadgeInProgress.vue

```diff
@@ -200,11 +176,11 @@ const onInteractCarousel = interaction => {
 	$kvTrackEvent('carousel', 'click-carousel-horizontal-scroll', interaction);
 };
 
-const badgeName = computed(() => props.badge?.fields?.challengeName ?? '');
+const badgeName = computed(() => props.badge?.challengeName ?? '');
 // TODO: Replace once the work of using the icon of the correct in-progress badge level is complete
 const badgeImage = computed(() => props.badge?.fields?.badgeImage?.fields?.file?.url ?? '');
 const badgeLevel = computed(() => props.badge?.fields?.level ?? '');
-const totalProgress = computed(() => props.badge?.totalProgressToAchievement ?? 0);
+const totalProgress = computed(() => props.badge?.achievementData?.totalProgressToAchievement ?? 0);
```

Comment:
Do we need reactive props here?

Reply:
Probably, since the modal can be re-opened with another badge.

Reply:
Same for other cases...

Reply:
Right, I'm assuming we generally want reactivity here for when the badge modal is re-opened with another badge.

Reply:
It's some upcoming work for that?

Reply:
What do you mean, these fields are already reactive, `computed` based on `props`?

Reply:
True! I mean if passing toRefs is needed...

Reply:
It shouldn't be needed, since we're not deconstructing the props.